chromium_src/third_party/blink/renderer/core/css/css_font_selector.cc

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/core/css/css_font_selector.h"
+#include "third_party/blink/public/platform/web_content_settings_client.h"
+#include "third_party/blink/renderer/core/execution_context/execution_context.h"
+
+#include "src/third_party/blink/renderer/core/css/css_font_selector.cc"
+
+namespace blink {
+
+bool CSSFontSelector::AllowFontFamily(const AtomicString& family_name) {
```

Comment:
instead of adding this as a method on CSSFontSelector, can we just pass in the execution context when we call it so we don't have to add the method to the header?

Reply:
in fact can't we just call BraveSessionCache::AllowFontFamily directly and pass it the context instead of the settings?

Reply:
hmm... I see the problem https://github.com/brave/brave-core/pull/12234/files#diff-96bfe3d4c355542196ef81bea85c80338b6290ed1d8ad75beaa514055997e578R7

Reply:
but can we just do this in `CSSFontSelectorBase` so it covers both? The implementations are identical, right?

Reply:
oh wait, this can work because `GetExecutionContext` is a public method on FontSelector